docs/conf.py

```diff
@@ -6,14 +6,38 @@
 
 from __future__ import annotations
 
-# Warning: do not change the path here. To use autodoc, you need to install the
-# package first.
+import sys
+import warnings
+from pathlib import Path
+
+if sys.version_info < (3, 8):
+    import importlib_metadata
+else:
+    import importlib.metadata as importlib_metadata
+
+ROOT = Path(__file__).parent.parent.resolve()
+
+
+try:
+    version = importlib_metadata.version("scikit-build-core")
+except ModuleNotFoundError:
+    warnings.warning(
+        "Package should be installed to produce documentation! Trying work around assuming a git archive was used."
+    )
+
+    from setuptools_scm import get_version
+
+    version = get_version(root=ROOT, fallback_root=ROOT)
+
+    sys.path.append(str(ROOT / "src"))
+
 
 # -- Project information -----------------------------------------------------
 
 project = "scikit-build-core"
 copyright = "2022, The Scikit-Build admins"
 author = "Henry Schreiner"
+release = ".".join(version.split(".")[:2])
```

Comment:
Check out the [generated documentation](https://scikit-build-core--372.org.readthedocs.build/en/372/). It cuts off at `0.4` instead of `0.4.5`. Maybe it is better to regex this one to [`(\d+\.\d+\.\d+)(.*)`](https://regex101.com/r/pf7ydg/1)

Reply:
I was thinking that the docs don't change that much between patch versions, so I did this, but maybe it's better do an extra digit - especially since the first one is just 0, so there's really only a "major" version for now. Okay, will change.